OracleGoldenGate/21c/bin/deployment-main.sh

```diff
@@ -182,11 +222,16 @@ function start_nginx() {
 ##  Termination handler
 ##
 function termination_handler() {
-    [[ -z "${ogg_pid}" ]] || {
-        kill  "${ogg_pid}"
-        unset    ogg_pid
+    [[ -f "${OGG_HOME}/ogg.pid" ]] && {
+        kill  $(cat "${OGG_HOME}/ogg.pid")
```

Comment:
Er. Why kill the Service Manager here?

Reply:
The OGG primary instance never receives the os level termination signal.  The original code trying to trap the OGG PID is actually only trapping the PID of the tail command.

Since the PID of the OGG instance is created inside the python script, it must be gathered there.  In order to issue the shutdown of OGG when the main PID 1 is signaled to exit by the orchestration layer (docker/k8s/etc) the trap then kills the child processes.  The PID of the OGG instance was never really being notified.

This correctly sends the exit signal to the OGG process and it shutsdown cleanly, rather than waiting the container timeout period and forcefully killing all remaining PIDs.

Reply:
I'm not sure what you mean by 'OGG primary instance.'  As far as I know, all processes in a Docker container are sent the SIGTERM signal when the container is terminated. Service Manager, in particular, shuts down pretty quickly when it receives SIGTERM.

I would like to better understand which processes are not responding in a timely fashion to SIGTERM.

Anyway, this change is not really part of enable user scripts, so it should be in a separate PR at a minumum.